pageserver/compaction/src/identify_levels.rs

```diff
@@ -308,6 +309,26 @@ mod tests {
         Ok(())
     }
 
+    #[tokio::test]
+    async fn repro_identify_levels_bails_too_ealy_if_partitioned_keyspace_same_lsn() -> anyhow::Result<()> {
+        tracing_subscriber::fmt::init(); // so that RUST_LOG=trace works
+        // It's sorting by end_lsn only, so, this `vec![]` is already sorted.
+        // (The sort implementation may shuffle the elements anyway, but, let's assume it doesn't.)
+        // The `identify_levels` loop will bails out at the first layer that is too large.
+        // , i.e., layer B. (log message "too large").
+        // That leaves layer C out of the level, even though it belongs to it.
```

Comment:
It will in fact leave out *all* the layers from the returned Level. That's correct. The layers overlap, so it must include either all of them, or none. Because B is too large, they are all left out.